### ISSUE TITLE
2.2

### ISSUE THREAD

**Issue by sonali-sr** [rebeccajohnson88/PPOL564_slides_activities#57]
### 2.2 Create a document-term matrix from the preprocessed press releases and to explore top words (5 points)

**A.** Use the `create_dtm` function I provide (alternately, feel free to write your own!) and create a document-term matrix using the preprocessed press releases; make sure metadata contains the following columns:` id`, `compound sentiment` column you added, and the `topics_clean` column.

**B.** Print the top 10 words for press releases with compound sentiment in the top 5% (so the most positive sentiment)

<img width="424" alt="2,2B" src="https://user-images.githubusercontent.com/90288860/197816590-bbfac595-89dc-4fd7-bdd0-d61b79b08dfe.PNG">

**C.** Print the top 10 words for press releases with compound sentiment in the bottom 5% (so the most negative sentiment). 
**_Hint:_** for these, remember the pandas quantile function from pset two.

<img width="393" alt="2,2C" src="https://user-images.githubusercontent.com/90288860/197816635-fa762578-c46b-406d-8b6f-66e672d14fb1.PNG">

**D.** Print the top 10 words for press releases in each of the three topics_clean

<img width="204" alt="2 2D" src="https://user-images.githubusercontent.com/90288860/197816835-cfeb05e5-271b-4526-a2af-fd24f8ea1dbf.PNG">

For steps B - D, to receive full credit, write a function `get_topwords` that helps you avoid duplicated code when you find top words for the different subsets of the data. There are different ways to structure it but one way is to feed it subsetted data (so data subsetted to one topic etc.) and for it to get the top words for that subset.

**Resources:**
- Here contains an example of applying the create_dtm function: [https://github.com/rebeccajohnson88/PPOL564_slides_activities/blob/main/activities/fall_22/solutions/09_textasdata_partII_topicmodeling_solution.ipynb](https://github.com/rebeccajohnson88/PPOL564_slides_activities/blob/main/activities/fall_22/solutions/09_textasdata_partII_topicmodeling_solution.ipynb)

**Comment by rebeccajohnson88** [comment]
> Are different count results acceptable here, Professor?

yep differences are fine since depend on preprocessing etc - thanks!
